File: python/Module6/get_object.py
import boto3
import os

s3_client = boto3.client('s3', region_name='us-east-1')

bucket = os.environ['MY_BUCKET']


def get_object(bucket_name, object_key):
    with open('airportsdownload.csv', 'wb') as f:
        s3_client.download_fileobj(bucket_name, object_key, f)




def head_object(bucket_name, object_key):
    response = s3_client.head_object(
        Bucket=bucket_name,
        Key=object_key,
    )

    print(response)


head_object(bucket, 'airports.csv')
get_object(bucket, 'airports.csv')

File: net/Module6/GetObject/GetObject.cs
using Amazon.S3;
using Amazon;
using Amazon.S3.Model;

namespace GetObject;
public class GetObject
{
    public static async Task Main(string[] args)
    {
        string bucketName = Environment.GetEnvironmentVariable("MY_BUCKET");
        string keyName = "airports.csv";

        IAmazonS3 s3Client = new AmazonS3Client(RegionEndpoint.USEast1);
        await ReadObjectDataAsync(s3Client, bucketName, keyName);
    }

  public static async Task<bool> ReadObjectDataAsync(IAmazonS3 client, string bucketName, string objectName)
    {
        var request = new GetObjectRequest
        {
            BucketName = bucketName,
            Key = objectName,
        };

        using GetObjectResponse response = await client.GetObjectAsync(request);

        try
        {
            await response.WriteResponseStreamToFileAsync($"C://Temp/\\{objectName}", true, CancellationToken.None);
            return response.HttpStatusCode == System.Net.HttpStatusCode.OK;
        }
        catch (AmazonS3Exception ex)
        {
            Console.WriteLine($"Error saving {objectName}: {ex.Message}");
            return false;
        }
    }
}
